chia/data_layer/data_layer_util.py

```diff
@@ -574,3 +574,42 @@ def marshal(self) -> Dict[str, Any]:
             "error": self.error,
             "fee": None if self.fee is None else int(self.fee),
         }
+
+
+@dataclasses.dataclass(frozen=True)
+class CancelOfferRequest:
+    trade_id: bytes32
+    # cancel on chain (secure) vs. just locally
+    secure: bool
```

Comment:
if `secure=False`, we'll need to delete our pending root locally.

Reply:
agreed, thanks, will address.